modules.py

```diff
@@ -288,4 +304,80 @@ def rebin(coords, wavelength, size, detector_scale):
 
     x = (coords[1]-size//2.)*detector_scale/scale_factor+(size//2.)
     y = (coords[0]-size//2.)*detector_scale/scale_factor+(size//2.)
-    return y, x
\ No newline at end of file
+    return y, x
+
+
+class PolyPSF(OpticalArray):
+    """
+    Polychromatic PSF class. Inherits from OpticalArray.
+    """
+    def __init__(self, band, spectral_type='A', size=505, scale=0.01):
+        super(PolyPSF, self).__init__(size, poly=True, scale=scale)
+        self.band = band
+        self.spectral_type = spectral_type
+        self._wavelength_contributions = None
+        self.sed_file = 'sed_%s0_fe0.txt' % spectral_type.lower()
+        self._x = None
+        self._flux = None
+        self.name = 'polychromatic_psf'
+        self._dist = self.dist
+
+    def get_sed(self):
+        """
+        Reads the SED profile from files and get the spectral energy distribution.
+        :return: self._x, self._flux : two lists, wavelength and its associated flux.
+        """
+        if self._x is not None and self._flux is not None:
+            return self._x, self._flux
+        han = open('SED/%s' % self.sed_file)
```

Comment:
have a look at the 'with' process for doing this.

  with open('SED/{}'.format(self.sed_file) as han:
      data = han.readlines()

no need to close as that is handled by the with statement.